src/graph/backend/dnnl/op_executable.hpp

```diff
@@ -2681,7 +2681,8 @@ struct sdpa_executable_t : public op_executable_t {
                 = op->get_input_value(1)->get_logical_tensor().dims[1];
         status_t s = create_sdpa_pd(sdpa_pd_, p_engine.get(), md_q.get(),
                 md_k.get(), md_v.get(), md_dst.get(), md_mask.get(), scale_dt,
-                is_invert_scale_, kv_head_number, is_causal_mask_, attr.get());
+                is_invert_scale_, kv_head_number,
+                is_causal_mask_ ? top_left : undefined, attr.get());
```

Comment:
I just had a try with benchdnn, this change will not affect the functionality of top-left causal mask.
The only concern is, just as Dima commented, the enum of causal mask type is in global namespace, which I guess need a change. And when it changes, I'll take another look at how it will affect Graph side, thanks!